Neos.Neos/Classes/FrontendRouting/EventSourcedFrontendNodeRoutePartHandler.php

```diff
@@ -228,22 +238,25 @@ public function matchWithParameters(&$requestPath, RouteParameters $parameters)
     /**
      * @param string $uriPath
      * @param DimensionSpacePoint $dimensionSpacePoint
-     * @return MatchResult
+     * @return MatchResult|false
      * @throws NodeNotFoundException
      */
     private function matchUriPath(
         string $uriPath,
         DimensionSpacePoint $dimensionSpacePoint,
         SiteDetectionResult $siteDetectionResult,
         ContentRepository $contentRepository
-    ): MatchResult {
+    ): MatchResult|false {
```

Comment:
Do we really need the union type here? We could also throw an exception?  Or make the return type nullable?